tests/test_subdomains.py

```diff
@@ -427,3 +429,69 @@ class MySubdomains2(SubDomainSet):
                              [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=np.int32)
 
         assert((np.array(f.data[:]+g.data[:]) == expected).all())
+
+    def test_issue_1761(self):
```

Comment:
Any test with N>1 ? Like N=3?

Reply:
it doesn't matter, you can put N=1000. It's irrelevant for this issue/test. The only important thing are the *eqns* , the SubDomainSet is arbitrary